src/App.tsx

```diff
@@ -14,11 +16,103 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  goods: string[];
+  selectedGoods: string[],
+};
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    goods: [...goodsFromServer],
+    selectedGoods: [],
+  };
+
+  isSelected = (item: string, selectedGoods: string[]) => {
+    if (!this.state.selectedGoods.includes(item)) {
+      this.setState({ selectedGoods: [...selectedGoods, item] });
+    } else {
+      selectedGoods.splice(selectedGoods.indexOf(item), 1);
```

Comment:
it is not good practice to modify parameters of the function. No need to pass `selectedGoods` at all, you can get it from `state`